optuna/terminator/erroreval.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+import abc
+
+import numpy as np
+
+from optuna._experimental import experimental_class
+from optuna.study.study import Study
+from optuna.trial import Trial
+from optuna.trial._state import TrialState
+
+
+_CROSS_VALIDATION_SCORES_KEY = "terminator:cv_scores"
+
+
+class BaseErrorEvaluator(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def evaluate(self, study: Study) -> float:
+        pass
+
+
+@experimental_class("3.2.0")
```

Comment:
[nit] Multiple experimental warnings are displayed when using TerminatorCallback.

```python
from optuna.terminator.callback import TerminatorCallback

terminator = TerminatorCallback()
# 
# /.../example.py:29: ExperimentalWarning: TerminatorCallback is experimental (supported from v3.2.0). The interface can change in the future.
#   terminator = TerminatorCallback()
# /.../callback.py:20: ExperimentalWarning: Terminator is experimental (supported from v3.2.0). The interface can change in the future.
#   self._terminator = terminator or Terminator()
# /.../terminator.py:29: ExperimentalWarning: CrossValidationErrorEvaluator is experimental (supported from v3.2.0). The interface can change in the future.
#   self._error_evaluator = error_evaluator or CrossValidationErrorEvaluator()
```

Do you think these warning messages should be reduced? Perhaps we could only display the experimental warning when the Terminator class is initialized or when the `optuna.terminator` module is imported.

Reply:
Good catch. This may cause a little negative impact for the user experience. Though, let us discuss as a follow-up task.